server/src/main/java/org/apache/uniffle/server/storage/local/StorageChoosingPolicy.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.server.storage.local;
+
+import org.apache.uniffle.server.ShuffleDataFlushEvent;
+import org.apache.uniffle.storage.common.Storage;
+
+public interface StorageChoosingPolicy<T extends Storage> {
+
+  T choose(ShuffleDataFlushEvent event, T... candidates);
```

Comment:
Could we have better name? What's the difference between `choose` and `select`? Should we have a unify style?

Reply:
Why do we use array as parameter? Collection seems better than array according to the book <<Effective Java>>.